funcx_endpoint/funcx_endpoint/endpoint/endpoint_manager.py

```diff
@@ -0,0 +1,365 @@
+            with open(endpoint_json, 'r') as fp:
+                self.logger.debug("Connection info loaded from prior registration record")
+                reg_info = json.load(fp)
+                endpoint_uuid = reg_info['endpoint_id']
+        elif not endpoint_uuid:
+            endpoint_uuid = str(uuid.uuid4())
+        return endpoint_uuid
+
+    def start_endpoint(self, name, endpoint_uuid, endpoint_config):
+        self.name = name
+        self.endpoint_config = endpoint_config
+
+        endpoint_dir = os.path.join(self.funcx_dir, self.name)
+        endpoint_json = os.path.join(endpoint_dir, 'endpoint.json')
+
+        if not os.path.exists(endpoint_dir):
+            print(f'Endpoint {self.name} is not configured!')
+            print('1. Please create a configuration template with:')
+            print(f'    funcx-endpoint configure {self.name}')
+            print('2. Update configuration')
+            print('3. Start the endpoint.')
+            return
+
+        # These certs need to be recreated for every registration
+        keys_dir = os.path.join(endpoint_dir, 'certificates')
+        os.makedirs(keys_dir, exist_ok=True)
+        client_public_file, client_secret_file = zmq.auth.create_certificates(keys_dir, "endpoint")
+        client_public_key, _ = zmq.auth.load_certificate(client_public_file)
+        client_public_key = client_public_key.decode('utf-8')
+
+        # This is to ensure that at least 1 executor is defined
+        if not self.endpoint_config.config.executors:
+            raise Exception(f"Endpoint config file at {endpoint_dir} is missing executor definitions")
+
+        funcx_client = FuncXClient(funcx_service_address=self.endpoint_config.config.funcx_service_address)
+
+        endpoint_uuid = self.check_endpoint_json(endpoint_json, endpoint_uuid)
+
+        self.logger.info(f"Starting endpoint with uuid: {endpoint_uuid}")
+
+        # Create a daemon context
+        # If we are running a full detached daemon then we will send the output to
+        # log files, otherwise we can piggy back on our stdout
+        if self.endpoint_config.config.detach_endpoint:
+            stdout = open(os.path.join(endpoint_dir, self.endpoint_config.config.stdout), 'w+')
+            stderr = open(os.path.join(endpoint_dir, self.endpoint_config.config.stderr), 'w+')
+        else:
+            stdout = sys.stdout
+            stderr = sys.stderr
+
+        try:
+            context = daemon.DaemonContext(working_directory=endpoint_dir,
+                                           umask=0o002,
+                                           pidfile=daemon.pidfile.PIDLockFile(
+                                               os.path.join(endpoint_dir, 'daemon.pid')),
+                                           stdout=stdout,
+                                           stderr=stderr,
+                                           detach_process=self.endpoint_config.config.detach_endpoint)
+
+        except Exception:
```

Comment:
Is this exception message accurate? Is it really ok for the execution to continue after this? It seems like a fatal error

Reply:
Added `sys.exit(-1)` for the exception